src/main/java/com/example/onlinebookstore/repository/category/CategoryRepository.java

```diff
@@ -0,0 +1,23 @@
+package com.example.onlinebookstore.repository.category;
+
+import com.example.onlinebookstore.model.Category;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.transaction.annotation.Transactional;
+
+public interface CategoryRepository extends JpaRepository<Category, Long> {
+    Optional<Category> findById(Long id);
+
+    @Modifying
+    @Transactional
+    @Query(value = "DELETE FROM books_categories WHERE book_id = :bookId", nativeQuery = true)
+    void removeCategoriesByBookId(Long bookId);
+
+    @Modifying
+    @Transactional
+    @Query(value = "DELETE FROM books_categories WHERE category_id = :categoryId",
+            nativeQuery = true)
+    void removeCategoriesByCategoryId(Long categoryId);
```

Comment:
This solution contradicts soft-delete policy used in project

If you use the @SQLDelete annotation to customize the SQL delete statement and the @Where annotation to apply a condition to exclude soft-deleted records on both the parent and child entities, you can leverage default Spring Data JPA repository methods without additional queries to automatically handle the soft-delete behavior. With these annotations in place on both the parent and child entities, you can use default Spring Data JPA repository methods, such as findById, findAll, and others, and they will automatically consider the soft-delete behavior. Deleted records will not be returned in query results, and you don't need to write additional queries to handle soft deletes explicitly.